src/univariate/continuous/normalcanon.jl

```diff
@@ -89,6 +89,12 @@ invlogccdf(d::NormalCanon, lp::Real) = xval(d, norminvlogccdf(lp))
 
 rand(rng::AbstractRNG, cf::NormalCanon) = cf.μ + randn(rng) / sqrt(cf.λ)
 
+rand!(rng::AbstractRNG, cf::NormalCanon, A::AbstractArray{<:Real}) =
+    # A .* inv(sqrt(cf.λ)) is faster but less accurate than A ./ sqrt(cf.λ)
+    # in floating point. This is a fast-math style optimization.
+    # It shouldn't be a problem for PRNG.
+    A .= muladd.(randn!(rng, A), inv(sqrt(cf.λ)), cf.μ)
```

Comment:
I think using `randn!` os an obvious improvement but I'm not sure whether `inv` + `muladd` and the inconsistency between `rand` and `rand!` is worth it (generally `inv` can easily lead to undesired promotions but I don't think that's a problem here). I'd use something like


```suggestion
function rand!(rng::AbstractRNG, cf::NormalCanon, A::AbstractArray{<:Real})
    randn!(rng, A)
    A .= cf.μ .+ A ./ sqrt(cf.λ)
    return A
end
```